From ebiederm@xmission.com  Wed May  5 16:10:08 2010
From: ebiederm@xmission.com (Eric W. Biederman)
Date: Wed, 05 May 2010 14:54:00 -0700
Subject: sysfs: Don't use enums in inline function declaration.
To: Greg KH <gregkh@suse.de>
Cc: Randy Dunlap <randy.dunlap@oracle.com>, Stephen Rothwell <sfr@canb.auug.org.au>
Message-ID: <m18w7y10cn.fsf_-_@fess.ebiederm.org>


It appears gcc can't cope with using an enum that is only declared in
an inline function declaration, that doesn't even use the variable
that is so declared.

Avoid the silliness and replace the enum with an int, and make gcc
happy.

Signed-off-by: Eric W. Biederman <ebiederm@aristanetworks.com>
Acked-by: Randy Dunlap <randy.dunlap@oracle.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/sysfs.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/sysfs.h
+++ b/include/linux/sysfs.h
@@ -328,7 +328,7 @@ static inline void sysfs_put(struct sysf
 {
 }
 
-static inline void sysfs_exit_ns(enum kobj_ns_type type, const void *tag)
+static inline void sysfs_exit_ns(int type, const void *tag)
 {
 }
 
